docs/concepts/native-interop.rst

```diff
@@ -0,0 +1,254 @@
+  DLL specified in ``DllImport`` attribute.
+
+The rest of the example is just invoking the method as you would any other managed 
+method. 
+
+Of course, the runtime allows communication to flow both ways 
+which enables you to call into managed artifacts from native functions, using 
+function pointers. The closest thing to a function pointer in managed code is a 
+**delegate**, so this is what is used to allow callbacks from native code into 
+managed code. 
+
+The way to use this feature is similar to managed to native process described 
+above. For a given callback, you define a delegate that matches the signature, 
+and pass that into the external method. The runtime will take care of everything 
+else.
+
+.. code-block:: c#
+    :linenos:
+
+    using System;
+    using System.Runtime.InteropServices;
+
+    namespace ConsoleApplication1 {
+
+        class Program {
+
+            delegate bool EnumWC(IntPtr hwnd, IntPtr lParam);
+
+            [DllImport("user32.dll")]
+            static extern int EnumWindows(EnumWC hWnd, IntPtr lParam);
+
+            static bool OutputWindow(IntPtr hwnd, IntPtr lParam) {
+                Console.WriteLine(hwnd.ToInt64());
+                return true;
+            }
+
+            static void Main(string[] args) {
+                EnumWindows(OutputWindow, IntPtr.Zero);
+            }
+        }
+    }
+    
+
+Before we walk through our example, it is good to go over the signatures of the 
+unmanaged functions we need to work with. The function we want to call to 
+enumerate all of the windows has the following signature:
+``BOOL EnumWindows (WNDENUMPROC lpEnumFunc, LPARAM lParam);``
+
+The first parameter is a callback. The said callback has the following signature: 
+``BOOL CALLBACK EnumWindowsProc (HWND hwnd, LPARAM lParam);``
+
+With this in mind, let's walk through the example:
+
+* Line #8 in the example defines a delegate that matches the signature of the 
+  callback from unmanaged code. Notice how the LPARAM and HWND types are 
+  represented using ``IntPtr`` in the managed code. 
+* Lines #10 and #11 introduce the ``EnumWindows`` function from the user32.dll 
+  library. 
+* Lines #13 - 16 implement the delegate. For this simple example, we just want 
+  to output the handle to the console.
+* Finally, in line #19 we invoke the external method and pass in the delegate.
+
+Both of the above examples depend on parameters, and in both cases, the parameters 
+are given as managed types. Runtime does the "right thing" and processes these 
+into its equivalents on the other side. Since this process is really important 
+to writing quality native interop code, let's take a look at what happens when 
+the runtime *marshals* the types. 
+
+Type marshalling
+^^^^^^^^^^^^^^^^
+**Marshalling** is the process of packing up types when they need to cross the 
+managed boundary into native and vice versa. 
+
+Why is this needed? Simply put, the types in the managed and unmanaged world 
```

Comment:
I like the wording in this article: https://msdn.microsoft.com/en-us/library/aa446538.aspx. 

"convert between the unmanaged and managed representations" although I wouldn't use the term "unmanaged".

I'd avoid terms like "simply put". Just say what you intend to say. "to its own devices" is a bit vague and makes it almost sound unpredictable. Instead, you should communicate that there is a default policy for interop. Also, we should be more vague on what is doing the interop. In CoreCLR and CLR, it is the runtime that does the interop. In .NET Native, all of the interop code is in the app, not the runtime. That's the direction we're headed.